doc/thread_pools.md

```diff
@@ -159,3 +159,20 @@ future = Future.new(:executor => pool).execute do
    #work
 end
 ~~~
+
+## Forking
+
+Some Ruby versions allow the Ruby process to be [forked](http://ruby-doc.org/core-2.3.0/Process.html#method-c-fork). Generally, mixing threading and forking is an [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern). Threading and forking are both concurrency techniques and mixing the two rarely provides a benefit. Moreover, Ruby does not copy any threads execept the main thread when forking. Thus threads created before the fork become unusable ("dead") in the forked process. This aspect of forking is a significant issue for any application or library which spawns threads, including but not limited to Concurrent Ruby. It is strongly advised that applications using `ThreadPoolExecutor` either directly or indirectly (via high-level concurrency abstractions like `Future` and `Actor`) do **not** also fork. Since Concurrent Ruby is a foundational library often used by gems which are in turn used by other applications it is impossible to prevent forking upstream. Concurrent Ruby therefore makes a few guaratnees about the behavior of `ThreadPoolExecutor` after forking.
+
+*Concurrent Ruby guarantees that any thread pools created on by the forking process before the fork remain functional on the forked process after the fork. It also guarantees that jobs post to a thread pool before a fork remain on the thread pool in the forking process after the fork.*
```

Comment:
"created on by"

---

Maybe "remain on the thread pool in (only) the forking process"?